src/Features/ERDDAP/hooks/TableDAPComponents.tsx

```diff
@@ -73,6 +76,42 @@ export const UseDatasets: React.FunctionComponent<UseDatasetsProps> = ({
     }
   }
 
+  if (loadedDatasets && platformId) {
+    const platforms: any = queryClient.getQueryData(["buoybarn-platforms"])
+    if (platforms) {
+      const platform = platforms.features.find((f) => f.id === platformId)
+      const latestReading = loadedDatasets.map((d) => {
+        return {
+          name: d.name,
+          latestValue: d.timeSeries[d.timeSeries.length - 1].reading,
+          time: d.timeSeries[d.timeSeries.length - 1].time.toISOString(),
+        }
+      })
+
+      const updatedReadings = platform.properties.readings.map((reading) => {
+        const update = latestReading.find((r) => r.name === reading.variable)
+        return update ? { ...reading, value: update.latestValue, time: update.time } : { ...reading }
+      })
+
+      const updatedPlatform = {
+        ...platform,
+        properties: {
+          ...platform.properties,
+          readings: updatedReadings,
+        },
+      }
+      const updatedPlatforms = {
+        ...platforms,
+        features: platforms.features.map((f) => (f.id === platformId ? updatedPlatform : f)),
+      }
+      const updatedValues = updatedReadings.map((r) => r.value).toString()
+      const originalValues = platform.properties.readings.map((r) => r.value).toString()
+      if (originalValues !== updatedValues) {
+        queryClient.setQueryData(["buoybarn-platforms"], updatedPlatforms)
+      }
+    }
+  }
```

Comment:
It might be worth some extra inline comments to document the intent of this code since it's both doing something cool, and not completely obvious.